modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccelerators.testcase

```diff
@@ -14,93 +14,334 @@ definition {
+				roleTitle = "guest");
+		}
+
+		task ("When the admin logs out and navigates to the Minium catalog page") {
+			User.logoutPG();
 
-		AssertElementNotPresent(locator1 = "CommerceAccelerators#GENERAL_ADD_TO_CART_BUTTON_ENABLED");
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/catalog");
+		}
+
+		task ("Then the quantity selector and the 'add to cart' button are disabled") {
+			AssertElementNotPresent(locator1 = "CommerceAccelerators#GENERAL_ADD_TO_CART_BUTTON_ENABLED");
 
-		AssertElementNotPresent(locator1 = "CommerceAccelerators#GENERAL_QUANTITY_SELECTOR_INPUT_FIELD_ENABLED");
+			AssertElementNotPresent(locator1 = "CommerceAccelerators#GENERAL_QUANTITY_SELECTOR_INPUT_FIELD_ENABLED");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-10783. The add to cart button and quantity selector are enabled for a logged user on B2C scenario"
+	@priority = 3
+	test AddToCartButtonAndQuantitySelectorAreEnabledOnB2CScenarioForLoggedUser {
+		property portal.acceptance = "false";
+
+		task ("Given a Minium site with a user without any roles is created") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+
+			JSONUser.addUser(
+				userEmailAddress = "testuser@liferay.com",
+				userFirstName = "test",
+				userLastName = "user",
+				userScreenName = "testuser");
+
+			JSONUser.setFirstPassword(
+				agreeToTermsAndAnswerReminderQuery = "true",
+				requireReset = "false",
+				userEmailAddress = "testuser@liferay.com");
+
+			JSONUser.addUserToSite(
+				groupName = "Minium",
+				userEmailAddress = "testuser@liferay.com");
+		}
+
+		task ("And a site type to 'B2C' for the Miniun Portal channel is set") {
+			CommerceChannels.openChannelsAdmin();
+
+			CommerceNavigator.gotoEntry(entryName = "Minium Portal");
+
+			Select(
+				key_fieldLabel = "Commerce Site Type",
+				locator1 = "Select#GENERIC_SELECT_FIELD",
+				value1 = "B2C");
+
+			Button.clickSave();
+		}
+
+		task ("When the user logs in") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "testuser@liferay.com");
+		}
+
+		task ("Then the quantity selector and the 'add to cart' button are enabled") {
+			AssertElementPresent(locator1 = "CommerceAccelerators#GENERAL_ADD_TO_CART_BUTTON_ENABLED");
+
+			AssertElementPresent(locator1 = "CommerceAccelerators#GENERAL_QUANTITY_SELECTOR_INPUT_FIELD_ENABLED");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-10783. The add to cart button and quantity selector are enabled for an unlogged user on B2C scenario"
+	@priority = 3
+	test AddToCartButtonAndQuantitySelectorAreEnabledOnB2CScenarioForUnloggedUser {
+		property portal.acceptance = "false";
+
+		task ("Given a Minium site is created") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+		}
+
+		task ("And a site type to 'B2C' for the Miniun Portal channel is set") {
```

Comment:
Please update this task as follows:
task ("And 'B2C' site type is set for the Minium Portal channel") {